tests/metadata_stores/test_provenance_golden_reference.py

```diff
@@ -229,6 +230,20 @@ def case_clickhouse(self, hash_algorithm: HashAlgorithm, clickhouse_db: str):
                 f"Hash algorithm {hash_algorithm} not supported by {ClickHouseMetadataStore}"
             )
 
+    @parametrize_with_cases("hash_algorithm", cases=HashAlgorithmCases)
+    def case_delta(self, hash_algorithm: HashAlgorithm, tmp_path):
```

Comment:
I think this metadata store should not be included in this test. It is not involved with provenance calculation. 

It should only be tested for metadata operations, e.g. read/write/delete. 

We should have a similar, standard test suite for this operations, and we should run them through all our metadata stores. 

This standard test suite should be created in another PR.